com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/scoping/InferredImplicitReferencesAdapter.java

```diff
@@ -28,27 +28,49 @@
 /**
  * Adapter which registers implicit references between two resources. Note that both the source and target objects of these references don't actually exist.
  */
-public class ImplicitReferencesAdapter extends AdapterImpl {
+public class InferredImplicitReferencesAdapter extends AdapterImpl {
 
   /** Constant URI fragment used for both source and target object of reference. */
   private static final String IMPLICIT_FRAGMENT = URI.encodeFragment("implicit!", false); //$NON-NLS-1$
 
+  /** Constant URI fragment used for both source and target object of reference. */
+  private static final String INFERRED_FRAGMENT = URI.encodeFragment("inferred!", false); //$NON-NLS-1$

```

Comment:
can you perhaps defined the public static constant 'inferred!' here and use it in the MonitoredClusteringBuilderState class?